GoogleDataLogger/GoogleDataLogger/Classes/GDLLogEvent.m

```diff
@@ -31,6 +31,31 @@ - (instancetype)initWithLogMapID:(NSString *)logMapID logTarget:(NSInteger)logTa
   return self;
 }
 
+- (instancetype)copy {
+  GDLLogEvent *copy = [[GDLLogEvent alloc] initWithLogMapID:_logMapID logTarget:_logTarget];
+  copy.extension = _extension;
+  copy.extensionBytes = _extensionBytes;
+  copy.qosTier = _qosTier;
+  copy.clockSnapshot = _clockSnapshot;
+  return copy;
+}
+
+- (NSUInteger)hash {
+  // This loses some precision, but it's probably fine.
```

Comment:
Precision should be irrelevant for a hash, right?

Reply:
I think it matters in a theoretical sense, if the hash value of two times coincidentally have the same first 32 bits' worth of digits. In reality, it's not likely to matter, and I had to make this precision loss explicit in order to avoid a warning/lib lint failure.